### PR TITLE
feat(templates): introduce React InstantSearch Hooks Native template

### PR DESCRIPTION
**Pull request by sarahdayan**
This introduces a template for React InstantSearch Hooks with React Native.

This is a minimal template with a search box and an infinite list of hits, similar to our existing template for React InstantSearch Native.

### CODE REVIEW

File: src/templates/React InstantSearch Hooks Native/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "ais-ecommerce-demo-app",
```
Comment (dhayab) [inline src/templates/React InstantSearch Hooks Native/package.json:2]:
```suggestion
  "name": "{{name}}",
```

File: src/templates/React InstantSearch Hooks Native/package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "eject": "expo eject"
+  },
+  "dependencies": {
+    "algoliasearch": "4.12.1",
+    "expo": "~44.0.0",
+    "expo-status-bar": "~1.2.0",
+    "instantsearch.js": "4.38.1",
+    "react": "17.0.1",
+    "react-dom": "17.0.1",
+    "react-instantsearch-hooks": "6.22.0",
```
Comment (dhayab) [inline src/templates/React InstantSearch Hooks Native/package.json:20]:
```suggestion
    "react-instantsearch-hooks": "{{libraryVersion}}",
```

**Review by Haroenv — APPROVED** [review]
all good, when dhaya's comments are addressed

File: src/templates/React InstantSearch Hooks Native/package.json (hunk 1 above)
Comment (sarahdayan) [reply src/templates/React InstantSearch Hooks Native/package.json]: Thanks! Addressed in 2a058ce.

File: src/templates/React InstantSearch Hooks Native/package.json (hunk 2 above)
Comment (sarahdayan) [reply src/templates/React InstantSearch Hooks Native/package.json]: Thanks! Addressed in 2a058ce.

**Review by dhayab — APPROVED** [review]

**Pull request merged**